dvc/path/__init__.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+from dvc.utils.compat import urlparse
+
+
+class BasePathInfo(object):
+    scheme = None
+
+    def __init__(self, path, url):
+        self.path = path
+        self.url = url
+
+    def __str__(self):
+        return self.url
+
+
+class SSHPathInfo(BasePathInfo):
+    scheme = "ssh"
+
+    def __init__(self, host, user, port, path, url):
+        super(SSHPathInfo, self).__init__(path, url)
+        self.host = host
+        self.user = user
+        self.port = port
+
+
+class HDFSPathInfo(BasePathInfo):
+    scheme = "hdfs"
```

Comment:
Should we use scheme from the corresponding remotes or vice-versa? Just to avoid nasty misspelling issues. Feels like remotes should use
```
scheme = HDFSPathInfo.scheme
```

Reply:
Ah, though it is a little more complicated with HTTP down below... Indeed.